File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-5.py
class Solution:
    def findMin(self, nums: List[int]) -> int:

        l, r = 0, len(nums) - 1
        mini = float('inf')

        while l < r:
            m = (l+r)//2
            mini = min(mini, nums[m])
            if nums[m] > nums[r]:
                l = m + 1
            else:
                r = m - 1

        return min(mini, nums[l])

File: Data Structures & Algorithms/two-integer-sum/submission-5.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        map<int, int> count;

        for (int i=0; i<nums.size(); i++){
            if (count.count(target - nums[i])){
                return {count[target-nums[i]], i};
            }
            else{
                count[nums[i]] = i;
            }
        }
    }
};
